pkg/validation/application_installation.go

```diff
@@ -185,6 +185,19 @@ func ValidateApplicationInstallationDelete(ctx context.Context, client ctrlrunti
 		return allErrs
 	}
 
+	labels := ai.GetLabels()
+	if labels != nil {
+		if _, exists := labels[appskubermaticv1.ApplicationManagedByLabel]; exists {
+			return field.ErrorList{
+				field.Invalid(
+					field.NewPath("metadata").Child("labels"),
+					labels,
+					fmt.Sprintf("Managed ApplicationInstallation cannot be deleted. To delete it, remove the %q label", appskubermaticv1.ApplicationManagedByLabel),
+				),
+			}
+		}
+	}
```

Comment:
This can be really problematic because this would disallow cilium `ApplicationInstallation` from being deleted when a cluster itself is marked for deletion.